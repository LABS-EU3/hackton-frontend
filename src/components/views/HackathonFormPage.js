import React from "react";
import HackathonForm from "../templates/HackathonForm";

const HackathonFormPage = () => (
  <HackathonForm 
    user="Mildred Pascal"
  />
);

export default HackathonFormPage;
