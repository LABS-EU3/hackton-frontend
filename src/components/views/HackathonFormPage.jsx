import React from "react";

import HackathonForm from "../templates/HackathonForm";
import { useParams } from "react-router-dom";
import { useSelector } from "react-redux";

const HackathonFormPage = () => {
  const { id } = useParams();
  const event = useSelector(state =>
    state.events.data.find(e => e.id === Number(id))
  );
  return <HackathonForm initialState={event} />;
};

export default HackathonFormPage;
