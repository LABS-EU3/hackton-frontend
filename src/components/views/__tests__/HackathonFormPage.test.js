import React from 'react'
import { Router, Route } from 'react-router';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react'
import HackathonFormPage from '../HackathonFormPage';
const history = createMemoryHistory()

const renderWithRouter = Component => render(
  <Router history={history}>
    <Route component={Component} />
  </Router>
)

describe("It should render HackathonSinglePage page correctly", () => {
  
    it("renders the HackathonSinglePage component correctly", () => {
      const render = () => renderWithRouter(HackathonFormPage)
      expect(render).toMatchSnapshot();
    });
  });