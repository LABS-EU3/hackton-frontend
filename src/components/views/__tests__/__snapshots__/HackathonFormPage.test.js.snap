// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render HackathonSinglePage page correctly renders the HackathonSinglePage component correctly 1`] = `[Function]`;
